sky/jobs/controller.py

```diff
@@ -754,7 +770,10 @@ async def _run_one_task(self, task_id: int, task: 'sky.Task') -> bool:
                 job_id=self._job_id,
                 task_id=task_id,
                 force_transit_to_recovering=force_transit_to_recovering,
-                callback_func=callback_func)
+                callback_func=callback_func,
+                code=recovery_code,
```

Comment:
What does a `recovery_code` mean here? Can we have a single data structure, e.g. Dataclass or `NamedTuple` to represent both the reason and the code (do we really need a code)?